### ISSUE TITLE
[Paginator] Accessibility audit

### ISSUE THREAD

**Issue by DanielleRameau** [sl-design-system/components#2514]
### Provide a general summary of the issue here

Do a11y audit for paginator, create new issues if there arte any findings.

### 👤 Your name

DR

**Comment by KatarzynaWM** [comment]
testing environment and tools:

Windows 11
Chrome version 140.0.7339.128
screen reader: NVDA
Color Contrast Analyser from TPGi

-----
[paginator/page size/Basic]

When navigating between numbers in the paginator using the keyboard arrows, the grey rectangle indicating the current position has too low a contrast ratio (1.1:1), making it very difficult to see. The contrast ratio should be at least 3:1.

<img width="641" height="305" alt="Image" src="https://github.com/user-attachments/assets/d99d2996-a2d9-46f6-85c8-74d64d87d055" />

**Comment by KatarzynaWM** [comment]
[paginator/page size/Basic]

If you choose the last item on the list, the space surrounding the number gets smaller (lower) in size than when choosing other options. When I checked it before the size was the same, but there was also a "tick" icon present, now gone (don't have screen shots).

<img width="349" height="64" alt="Image" src="https://github.com/user-attachments/assets/4fd33ad6-2f09-4e00-9662-fca15d237236" />

<img width="285" height="74" alt="Image" src="https://github.com/user-attachments/assets/9646eda0-4ef6-4ffa-bcae-192cea1e6d7a" />

**Comment by KatarzynaWM** [comment]
[paginator/paginator/Basic]
